crates/vault-registry/src/lib.rs

```diff
@@ -641,10 +643,29 @@ impl<T: Config> Pallet<T> {
             Self::is_allowed_to_withdraw_collateral(vault_id, amount)?,
```

Comment:
Maybe something for the future, but this call seems to be loading the vault struct a couple of times.

Reply:
Should I add a parameter of type `Option<DefaultVault<T>>` to both functions?